src/sonic-config-engine/sonic_device_util.py

```diff
@@ -48,7 +48,19 @@ def get_system_mac():
     version_info = get_sonic_version_info()
 
     if (version_info['asic_type'] == 'mellanox'):
-        get_mac_cmd = "sudo decode-syseeprom -m"
+        # With Mellanox ONIE release(2019.05-5.2.0012) and above
+        # "onie_base_mac" was added to /host/machine.conf:
+        # onie_base_mac=e4:1d:2d:44:5e:80
+        # So we have another way to get the mac address besides decode syseeprom
+        # By this can mitigate the dependency on the hw-management service
+        base_mac_key = "onie_base_mac"
+        machine_vars = get_machine_info()
+        if machine_vars is not None and base_mac_key in machine_vars:
+            mac = machine_vars[base_mac_key]
+            mac = mac.strip()
+            return mac
```

Comment:
Do you need some validation before `return`, and fallback to old way if the new way failed?

Reply:
In function get_machine_info() it ruled out empty string case, here if we want to add a further validation I think I can use regular expressions to check whether it's a valid mac. and this check can also be applied to the mac fetched on old ways, what do you think?

Reply:
It's better to validate both new way and old way.

Reply:
@qiluo-msft please check the latest commit with mac validation.